lib/compiler.js

```diff
@@ -505,7 +509,12 @@ function resolveRelativePaths(relativePaths, appRootDir) {
   relativePaths.forEach(function(relativePath, k) {
     var start = relativePath.substring(0, 2);
     if (start === './' || start === '..') {
-      relativePaths[k] = path.resolve(appRootDir, relativePath);
+      var resolvedPath = resolveAppPath(appRootDir, relativePath);
+      if (resolvedPath !== undefined) {
+        relativePaths[k] = resolvedPath;
+      } else {
+        debug ('skipping boot script %s - unknown file', relativePath);
+      }
```

Comment:
Is this covered by a unit test too?

Also I think you can remove the check `if (start === './' || start === '..') {`now, because `resolveAppPath` correctly handles non-relative paths too, or not? Please add a unit-test for this too.